File: induscore/src/main/java/com/induscore/dto/agent/AgentDtos.java
package com.induscore.dto.agent;

import jakarta.validation.Valid;
import jakarta.validation.constraints.DecimalMax;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;

import java.util.List;
import java.util.Map;

/**
 * Agent 模块 DTO 聚合，便于前后端按统一契约联调。
 */
public final class AgentDtos {
    private AgentDtos() {
    }

    public record ChatMessage(
            @NotBlank(message = "role 不能为空") String role,
            @NotBlank(message = "content 不能为空") String content
    ) {
    }

    public record ChatContext(
            String scene,
            String page,
            Map<String, Object> pageParams,
            Long taskId,
            String idempotencyKey,
            String timeRange
    ) {
    }

    public record ChatOptions(
            Boolean stream,
            @Min(value = 64, message = "maxTokens 不能小于 64")
            @Max(value = 4096, message = "maxTokens 不能大于 4096")
            Integer maxTokens,
            @DecimalMin(value = "0.0", message = "temperature 不能小于 0")
            @DecimalMax(value = "1.0", message = "temperature 不能大于 1")
            Double temperature
    ) {
    }

    public record ChatRequest(
            String sessionId,
            List<@Valid ChatMessage> messages,
            @Valid ChatMessage message,
            @Valid ChatContext context,
            @Valid ChatOptions options,
            Boolean stream
    ) {
    }

    public record ToolCall(
            String tool,
            boolean success,
            long latencyMs
    ) {
    }

    public record Usage(
            int promptTokens,
            int completionTokens,
            int totalTokens
    ) {
    }

    public record ChatResponse(
            String sessionId,
            String answer,
            List<ToolCall> toolCalls,
            String traceId,
            Usage usage
    ) {
    }

    public record SessionSummary(
            String sessionId,
            String title,
            String lastMessageAt,
            String clientType
    ) {
    }

    public record SessionPage(
            List<SessionSummary> items,
            long total,
            int page,
            int pageSize
    ) {
    }

    public record MessageItem(
            String role,
            String content,
            String timestamp
    ) {
    }

    public record SessionDetail(
            String sessionId,
            String title,
            List<MessageItem> messages,
            long total,
            int page,
            int pageSize
    ) {
    }

    public record ArchiveResponse(
            String sessionId,
            String status
    ) {
    }

    public record DebugSession(
            String sessionId,
            String title,
            String status,
            String clientType,
            String createdAt,
            String updatedAt
    ) {
    }

    public record DebugMessage(
            Long id,
            String role,
            String content,
            Integer tokenCount,
            String traceId,
            String createdAt
    ) {
    }

    public record DebugToolCall(
            Long id,
            String traceId,
            String toolName,
            boolean success,
            String errorCode,
            Integer latencyMs,
            String inputJson,
            String outputJson,
            String createdAt
    ) {
    }

    public record DebugAudit(
            Long id,
            String traceId,
            String responseStatus,
            String requestSummary,
            String modelName,
            Integer promptTokens,
            Integer completionTokens,
            Integer totalTokens,
            String createdAt
    ) {
    }

    /**
     * 只读排查聚合视图：
     * - session: 会话主信息
     * - messages: 会话消息（按时间升序）
     * - toolCalls: 工具调用（可按 trace 过滤）
     * - audits: 审计日志（可按 trace 过滤）
     */
    public record DebugSnapshot(
            String filterSessionId,
            String filterTraceId,
            DebugSession session,
            List<DebugMessage> messages,
            List<DebugToolCall> toolCalls,
            List<DebugAudit> audits
    ) {
    }
}

File: induscore/android-app/app/src/main/java/com/induscore/mobile/data/remote/dto/AgentDtos.kt
package com.induscore.mobile.data.remote.dto

data class AgentChatMessage(
    val role: String,
    val content: String
)

data class AgentChatContext(
    val scene: String? = null,
    val page: String? = null,
    val pageParams: Map<String, Any?>? = null,
    val taskId: Long? = null,
    val idempotencyKey: String? = null,
    val timeRange: String? = null
)

data class AgentChatOptions(
    val stream: Boolean? = false,
    val maxTokens: Int? = null,
    val temperature: Double? = null
)

data class AgentChatRequest(
    val sessionId: String? = null,
    val messages: List<AgentChatMessage> = emptyList(),
    val message: AgentChatMessage,
    val context: AgentChatContext? = null,
    val options: AgentChatOptions? = null,
    val stream: Boolean? = false
)

data class AgentToolCall(
    val tool: String? = null,
    val success: Boolean = true,
    val latencyMs: Long = 0
)

data class AgentUsage(
    val promptTokens: Int = 0,
    val completionTokens: Int = 0,
    val totalTokens: Int = 0
)

data class AgentChatResponse(
    val sessionId: String,
    val answer: String,
    val toolCalls: List<AgentToolCall> = emptyList(),
    val traceId: String? = null,
    val usage: AgentUsage? = null
)

data class AgentSessionSummaryDto(
    val sessionId: String,
    val title: String? = null,
    val lastMessageAt: String? = null,
    val clientType: String? = null
)

data class AgentSessionPageDto(
    val items: List<AgentSessionSummaryDto> = emptyList(),
    val total: Long = 0,
    val page: Int = 1,
    val pageSize: Int = 20
)

data class AgentMessageItemDto(
    val role: String,
    val content: String,
    val timestamp: String? = null
)

data class AgentSessionDetailDto(
    val sessionId: String,
    val title: String? = null,
    val messages: List<AgentMessageItemDto> = emptyList(),
    val total: Long = 0,
    val page: Int = 1,
    val pageSize: Int = 50
)

data class AgentArchiveResponseDto(
    val sessionId: String,
    val status: String
)
